lib/foreman_tasks/tasks/export_tasks.rake

```diff
@@ -218,35 +231,37 @@ namespace :foreman_tasks do
       end
     end
 
+    filter = ENV['FILTER'] || "result != success && started_at > \"#{60.days.ago.to_s(:db)}\""
+    export_filename = ENV['FILE'] || "/tmp/task-export-#{DateTime.now.to_i}"
+    format =  ENV['FORMAT'] || 'html'
 
-    if ENV['tasks'] == 'all'
-      tasks = ForemanTasks::Task
-    else
-      tasks = ForemanTasks::Task.where("result != 'success'")
-    end
-
-    days = ENV['days'].try(:to_i) || 60
-    tasks = tasks.where('started_at > ?', days.days.ago)
-    export_filename = ENV['export'] || "/tmp/task-export-#{DateTime.now.to_i}.tar.gz"
+    tasks = ForemanTasks::Task.search_for(filter)
+    export_filename += (format == 'csv' ? '.csv' : '.tar.gz')
```

Comment:
I would suggest to add the suffix just in case the file name is autogenerated. It feels strange to pass `FILE=my_file.csv` and the data exported in `my_file.csv.csv`